packages/rs-dapi-client/src/platform/get_identity.rs

```diff
@@ -0,0 +1,112 @@
+//! `GetIdentity*` requests.
+
+use dapi_grpc::platform::v0::{self as platform_proto, Proof, ResponseMetadata};
+
+use super::IncompleteMessage;
+use crate::{transport::TransportRequest, DapiRequest, Settings};
+
+/// Request Identity bytes.
+#[derive(Debug)]
+pub struct GetIdentity {
+    /// Identity ID to search.
+    pub id: Vec<u8>,
+}
+
+/// DAPI response for [GetIdentity].
+#[derive(Debug)]
+pub struct GetIdentityResponse {
+    /// Serialized Identity
+    pub identity_bytes: Option<Vec<u8>>,
+    /// Response metadata
+    pub metadata: ResponseMetadata,
+}
+
+impl DapiRequest for GetIdentity {
+    type DapiResponse = GetIdentityResponse;
+
+    const SETTINGS_OVERRIDES: Settings = Settings::default();
+
+    type Error = IncompleteMessage;
+
+    type TransportRequest = platform_proto::GetIdentityRequest;
+
+    fn to_transport_request(&self) -> Self::TransportRequest {
+        platform_proto::GetIdentityRequest {
+            id: self.id.clone(),
+            prove: false,
+        }
+    }
+
+    fn try_from_transport_response(
+        transport_response: <Self::TransportRequest as TransportRequest>::Response,
+    ) -> Result<Self::DapiResponse, Self::Error> {
+        use platform_proto::get_identity_response::Result as GrpcResponseBody;
+        use platform_proto::GetIdentityResponse as GrpcResponse;
+
+        match transport_response {
+            GrpcResponse {
+                result: None,
+                metadata: Some(metadata),
+            } => Ok(GetIdentityResponse {
+                metadata,
+                identity_bytes: None,
+            }),
+            GrpcResponse {
+                result: Some(GrpcResponseBody::Identity(identity_bytes)),
+                metadata: Some(metadata),
+            } => Ok(GetIdentityResponse {
+                identity_bytes: Some(identity_bytes),
+                metadata,
+            }),
+            _ => Err(IncompleteMessage),
+        }
+    }
+}
+
+/// Request Identity bytes wrapped into proof.
+#[derive(Debug)]
+pub struct GetIdentityProof {
+    /// Identity ID to search.
+    pub id: Vec<u8>,
+}
+
+/// DAPI response for [GetIdentity].
+#[derive(Debug)]
+pub struct GetIdentityProofResponse {
+    /// Proof data that wraps Identity
+    pub proof: Proof,
+    /// Response metadata
+    pub metadata: ResponseMetadata,
+}
+
+impl DapiRequest for GetIdentityProof {
+    type DapiResponse = GetIdentityProofResponse;
+
+    const SETTINGS_OVERRIDES: Settings = Settings::default();
+
+    type Error = IncompleteMessage;
+
+    type TransportRequest = platform_proto::GetIdentityRequest;
+
+    fn to_transport_request(&self) -> Self::TransportRequest {
+        platform_proto::GetIdentityRequest {
+            id: self.id.clone(),
+            prove: true,
+        }
+    }
+
+    fn try_from_transport_response(
```

Comment:
Do we want to expose these *Proof funcs to users?

My opinion is that we should **always** request proofs, and verify them before returning to the user.
As part of verification, we will retrieve data from the proof, so that the user will get raw data that has already been verified.

In other words, I see proof verification to be just a part of transport layer, similar to how certificates are verified by all tls libraries out there.

To achieve that, you can use FromProof trait implementations done as part of light client.

Reply:
Yes that way my initial thought, but I was instructed not to do it that way and return just bytes and the rest should be handled outside of this library, but it would be logical to have a full circle 

Since I'm trying to support the same set of operations as js-dapi-client, which allows to pass `prove` parameter through proto, I split it into separate requests, because they return totally different things